packages/ui/src/Customization/CustomizationsExtensionDetail.tsx

```diff
@@ -1,14 +1,249 @@
+  i18nUsage: string;
+  i18nUsageMessage: string;
+  onDelete: (extensionName: string) => void;
+  onSelectIntegration: (integrationId: string) => void;
+  onUpdate: (extensionName: string) => void;
+  propDescription: string;
+  propName: string;
+  usedByIntegrations?: Integration[];
 }
 
 export class CustomizationsExtensionDetail extends React.Component<
   IExtensionDetailProps
 > {
+  public constructor(props: IExtensionDetailProps) {
+    super(props);
+    this.onDelete = this.onDelete.bind(this);
+    this.onUpdate = this.onUpdate.bind(this);
+  }
+
+  public createCardBody() {
+    return (
+      <>
+        {this.createOverviewSection()}
+        {this.createSupportsSection()}
+        {this.createUsageSection()}
+      </>
+    );
+  }
+
+  public createCardHeading() {
+    return (
+      <CardTitle>
+        <Row>
+          <h1 className={'col-sm-8 extension-title'}>
+            {this.props.extension.name}
+            {this.props.extension.extensionId ? (
+              <span className={'extension-id'}>{this.props.i18nIdMessage}</span>
+            ) : null}
+          </h1>
+          <div className="col-sm-4 text-right">
+            <OverlayTrigger overlay={this.getUpdateTooltip()} placement="top">
+              <Button bsStyle="primary" onClick={this.onUpdate}>
+                {this.props.i18nUpdate}
+              </Button>
+            </OverlayTrigger>
+            <OverlayTrigger overlay={this.getDeleteTooltip()} placement="top">
+              <Button
+                bsStyle="default"
+                disabled={
+                  OptionalIntUtils.getValue(this.props.extension.uses) !== 0
+                }
+                onClick={this.onDelete}
+              >
+                {this.props.i18nDelete}
+              </Button>
+            </OverlayTrigger>
+          </div>
+        </Row>
+      </CardTitle>
+    );
+  }
+
+  public createOverviewSection() {
+    return (
+      <div className="extension-group">
+        <h3>{this.props.i18nOverview}</h3>
+        <dl className="dl-horizontal">
+          <dt>{this.props.i18nName}</dt>
+          <dd>{this.props.extension.name}</dd>
+          <dt>{this.props.i18nDescription}</dt>
+          <dd>
+            {this.props.extension.description
+              ? this.props.extension.description
+              : ''}
+          </dd>
+          <dt>{this.props.i18nType}</dt>
+          <dd>{this.props.i18nTypeMessage}</dd>
+          <dt>{this.props.i18nLastUpdate}</dt>
+          <dd>{this.props.i18nLastUpdateDate}</dd>
+        </dl>
+      </div>
+    );
+  }
+
+  public createSupportsSection() {
+    let supportedMsg;
+
+    if (this.props.extension.extensionType === 'Steps') {
+      supportedMsg = this.props.i18nSupportedSteps;
+    } else if (this.props.extension.extensionType === 'Connectors') {
+      supportedMsg = this.props.i18nSupportedConnectors;
+    }
+
+    return (
+      <div className="extension-group">
+        <h3>{supportedMsg}</h3>
```

Comment:
What's shown here for a "library" extension?

Reply:
Good catch. In the angular version the `h3` is not added for library extension. I will change so that the `h3` is not added for this case.